app/src/utils/data/contributions.ts

```diff
@@ -368,3 +375,103 @@ export function grantDonationListener(
     off: () => grantRoundManager.value.off(filter, listener),
   };
 }
+
+/**
+ * @notice given a userAddress and the contributions, return only the contributions for that address
+ * @param userAddress
+ * @param contributions
+ */
+export function filterContributionsByUserAddress(userAddress: string, contributions: Contribution[]) {
+  if (contributions && userAddress) {
```

Comment:
`Boolean([])` resolves to true, so you need to explicitly check array length here if you want to return `undefined` for an empty array

```suggestion
  if (contributions.length > 0 && userAddress) {
```